apps/app/src/views/routes/Order.tsx

```diff
@@ -339,354 +373,446 @@ export const Order = () => {
+              isDisabled={
+                loading ||
+                order.state === types.OrderState.Canceled ||
+                order.state === types.OrderState.Completed ||
+                order?.fulfillment?.state === 'SHIPPED'
+              }
+              onClick={async () => {
+                const decision = await confirmWrapper('Cancel this order?', {
+                  description: (
+                    <RadioGroup onChange={setCancelReason}>
+                      <Text mb={2}>Please select a reason for canceling</Text>
+                      <Stack direction="column">
+                        {cancelReasons.map((reason) => (
+                          <Radio key={reason} value={reason}>
+                            {reason}
+                          </Radio>
+                        ))}
+                      </Stack>
+                    </RadioGroup>
+                  ),
+                  cancelText: "No, Don't Cancel",
+                  confirmText: 'Yes, Cancel',
+                  darkMode: colorMode !== 'light',
+                  colorScheme: 'red'
+                });
+                if (decision) {
+                  setUpdating(true);
+                  const variables = {
+                    id,
+                    ...(cancelReasonRef.current && { reason: cancelReasonRef.current })
+                  };
+                  await cancelOrder({ variables });
+                  setUpdating(false);
+                }
+              }}
+            >
+              Cancel Order
+            </Button>
+            <Button
+              aria-label="Report Issue"
+              colorScheme="blue"
+              onClick={() => setIsTicketModalOpen(true)}
+              isDisabled={loading}
+            >
+              Report Issue
+            </Button>
+          </HStack>
+        }
+      >
+        <Card>
+          <CardHeader>
             <Stack
               direction={{ base: 'column', md: 'row' }}
-              align={{ base: 'start', md: 'stretch' }}
-              spacing={2}
+              justify="space-between"
+              align="center"
+              width="full"
             >
-              <Text fontWeight="medium">
+              <Text fontWeight="medium" flex="1">
                 {loading ? <Skeleton height="30px" width="250px" /> : formatFills(order.fills)}
               </Text>
               {loading ? (
                 <Skeleton width="70px" height="24px" borderRadius="xl" />
               ) : (
-                <OrderStatusBadge
-                  fulfillmentState={order.fulfillment?.state}
-                  orderState={order.state}
-                />
+                <Stack flexShrink={0}>
+                  <OrderStatusBadge
+                    fulfillmentState={order.fulfillment?.state}
+                    orderState={order.state}
+                  />
+                </Stack>
               )}
+
+              <CopyText size="xs" text={order?.id} />
```

Comment:
should this should if `order?.id` is null?

Reply:
The component returns null
```
if (!text) return null;
```